jelatyna-backend/src/main/java/pl/confitura/jelatyna/presentation/PresentationController.java

```diff
@@ -72,27 +85,20 @@
     @PostMapping("/presentations/{presentationId}/cospeakers/{email:.+}")
     @Transactional
     public ResponseEntity<?> addCospeaker(@PathVariable String presentationId, @PathVariable String email) {
-        User user = this.userRepository.findByEmail(email);
+        User user = this.userFacade.findByEmail(email);
         if (user == null) {
             return ResponseEntity.notFound().build();
         }
 
         Presentation presentation = this.repository.findById(presentationId);
         if (presentation.isOwnedBy(email)) {
-            return ResponseEntity.status(HttpStatus.CONFLICT).body("You cannot add yourself as a speaker!");
-        }
-        if (presentation.hasCospeaker(email)) {
             return ResponseEntity.status(HttpStatus.CONFLICT).body("This speaker is already added to this presentation");
         }
-        presentation.getSpeakers().add(user);
-        return ResponseEntity.ok(user);
-    }
-
-    private Set<User> removeCospeakerByEmail(String email, Set<User> cospeakers) {
-        return cospeakers.stream().filter(it -> !it.getEmail().equalsIgnoreCase(email)).collect(Collectors.toSet());
+        presentation.getSpeakers().add(Speaker.fromUser(user));
```

Comment:
czy ta cała logika nie powinna wylądować w fasadzie? a tutaj tylko mapping odpowiedzi na RESTowy interface?

Reply:
to w sumie jest ogólny komentarz do wszystkich metod w controllerze które mają jakąs taką bardziej skomplikowaną logikę.

Jeśli byśmy chcieli np wprowadzić GraphQL, to ta logika musiała by być skopiowana.

Reply:
tak, dzieki

Reply:
wiem już czemu tak nie zrobiłem :)
planowałem to przy refactoringu "modułu" prezentacji
w tym PR skupiłem się na User

Reply:
oki, spoczko.